File: javasample/src/main/java/com/github/stephenvinouze/advancedrecyclerview/javasample/fragments/GestureSectionRecyclerFragment.java
package com.github.stephenvinouze.advancedrecyclerview.javasample.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.github.stephenvinouze.advancedrecyclerview.core.enums.ChoiceMode;
import com.github.stephenvinouze.advancedrecyclerview.gesture.extensions.GestureCallback;
import com.github.stephenvinouze.advancedrecyclerview.gesture.extensions.GestureKt;
import com.github.stephenvinouze.advancedrecyclerview.javasample.adapters.SampleSectionAdapter;
import com.github.stephenvinouze.advancedrecyclerview.javasample.models.Sample;

import java.util.ArrayList;
import java.util.Collections;

/**
 * Created by Stephen Vinouze on 06/11/2015.
 */
public class GestureSectionRecyclerFragment extends AbstractRecyclerFragment {

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        final SampleSectionAdapter sectionAdapter = new SampleSectionAdapter(getContext());
        sectionAdapter.setChoiceMode(ChoiceMode.MULTIPLE);
        sectionAdapter.setOnClick((view1, position) -> {
            Sample sample = sectionAdapter.getItems().get(position);
            Toast.makeText(getActivity(), "Item clicked : " + sample.getId() + " (" + sectionAdapter.getSelectedItemViewCount() + " selected)", Toast.LENGTH_SHORT).show();
            return null;
        });

        ArrayList<Sample> samples = Sample.mockItems();
        Collections.sort(samples, (lhs, rhs) -> lhs.getRate() - rhs.getRate());

        sectionAdapter.setItems(samples);

        recyclerView.setAdapter(sectionAdapter);
        GestureKt.enableGestures(
                recyclerView,
                ItemTouchHelper.UP | ItemTouchHelper.DOWN,
                ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT,
                new GestureCallback() {
                    @Override
                    public boolean onMove(int fromPosition, int toPosition) {
                        Toast.makeText(getContext(), "Item moved from position " + fromPosition + " to position " + toPosition, Toast.LENGTH_SHORT).show();
                        return false;
                    }

                    @Override
                    public void onSwiped(int position, int direction) {
                        Toast.makeText(getContext(), "Item swiped at position " + position, Toast.LENGTH_SHORT).show();
                    }
                }
        );
    }

}

File: sample/src/main/java/com/github/stephenvinouze/advancedrecyclerview/sample/fragments/GestureSectionRecyclerFragment.kt
package com.github.stephenvinouze.advancedrecyclerview.sample.fragments

import android.os.Bundle
import androidx.recyclerview.widget.ItemTouchHelper
import android.view.View
import android.widget.Toast
import com.github.stephenvinouze.advancedrecyclerview.core.enums.ChoiceMode
import com.github.stephenvinouze.advancedrecyclerview.gesture.extensions.enableGestures
import com.github.stephenvinouze.advancedrecyclerview.sample.adapters.SampleSectionAdapter
import com.github.stephenvinouze.advancedrecyclerview.sample.models.Sample
import kotlinx.android.synthetic.main.recycler_layout.*

/**
 * Created by Stephen Vinouze on 06/11/2015.
 */
class GestureSectionRecyclerFragment : AbstractRecyclerFragment() {

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val sectionAdapter = SampleSectionAdapter().apply {
            items = Sample.mockItems().sortedBy { it.rate }.toMutableList()
            choiceMode = ChoiceMode.MULTIPLE
            onClick = { _, position ->
                val sample = items[position]
                Toast.makeText(context, "Item clicked : ${sample.id} ($selectedItemViewCount selected)", Toast.LENGTH_SHORT).show()
            }
        }

        recyclerView.adapter = sectionAdapter
        recyclerView.enableGestures(
                dragDirections = ItemTouchHelper.UP or ItemTouchHelper.DOWN,
                swipeDirections = ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT,
                onMove = { fromPosition, toPosition ->
                    Toast.makeText(context, "Item moved from position $fromPosition to position $toPosition", Toast.LENGTH_SHORT).show()
                    true
                },
                onSwipe = { position, _ ->
                    Toast.makeText(context, "Item swiped at position $position", Toast.LENGTH_SHORT).show()
                }
        )
    }

}
